onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -157,12 +163,15 @@
   // When DXCore APIs are available QI for relevant enumeration interfaces
   constexpr bool use_dxcore_workload_enumeration = false;
   if (!use_dxcore_workload_enumeration) {
-    // Get a list of all the adapters that support compute
-    GUID attributes[]{ DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE };
     ORT_THROW_IF_FAILED(
-      adapter_factory->CreateAdapterList(_countof(attributes),
```

Comment:
Shouldnt this old code stay the same, and instead of using Generic_ML the CreateAdapterListByWorkload API should be called?

Reply:
Discussed and we'll do this in a separate change.  This requires extra interesting testing.